modules/misc/src/InternalVolume.C

```diff
@@ -50,16 +52,68 @@ InternalVolume::InternalVolume(const InputParameters & parameters) :
 // That is, the volume of the domain is the integral over the surface of the domain
 // of the x position of the surface times the x-component of the normal of the
 // surface.
-//
+
+void
+InternalVolume::initialSetup()
+{
+  SideIntegralPostprocessor::initialSetup();
+
+  const std::set<SubdomainID> & subdomains = _mesh.meshSubdomains();
+  std::set<SubdomainID>::const_iterator iter = subdomains.begin();
+  const std::set<SubdomainID>::const_iterator iter_end = subdomains.end();
+  for ( ; iter != iter_end; ++iter)
+  {
+    const std::set<BoundaryID> & boundaries = _mesh.getSubdomainBoundaryIds(*iter);
+    std::set<BoundaryID>::const_iterator bnd = boundaries.begin();
+    const std::set<BoundaryID>::const_iterator bnd_end = boundaries.end();
+    for ( ; bnd != bnd_end; ++bnd )
+    {
+      const std::set<BoundaryID> & b = boundaryIDs();
+      std::set<BoundaryID>::const_iterator bit = b.begin();
+      const std::set<BoundaryID>::const_iterator bit_end = b.end();
+      for ( ; bit != bit_end; ++bit )
+        if (*bit == *bnd)
+        {
+          Moose::CoordinateSystemType coord_sys = _fe_problem.getCoordSystem(*iter);
+          if (_component != 0 && coord_sys == Moose::COORD_RSPHERICAL)
```

Comment:
if (*bit == *bnd) then one of the boundaries this object is restricted to matches one of the boundaries of the block *iter.  It looks to me like the check is restricted like you mentioned.